metadata-ingestion/src/datahub/ingestion/source/looker/lookml_source.py

```diff
@@ -669,6 +673,184 @@ def load_viewfile(
+    connection_definition: LookerConnectionDefinition
+    source_config: LookMLSourceConfig
+    reporter: LookMLSourceReport
+    refinement_cache: Dict[
+        str, dict
+    ]  # looker view-name vs it is original view with applied refinement
+
+    def __init__(
+        self,
+        looker_model: LookerModel,
+        looker_viewfile_loader: LookerViewFileLoader,
+        connection_definition: LookerConnectionDefinition,
+        source_config: LookMLSourceConfig,
+        reporter: LookMLSourceReport,
+    ):
+        self.looker_model = looker_model
+        self.looker_viewfile_loader = looker_viewfile_loader
+        self.connection_definition = connection_definition
+        self.source_config = source_config
+        self.reporter = reporter
+        self.refinement_cache = {}
+
+    @staticmethod
+    def is_refinement(view_name: str) -> bool:
+        return view_name.startswith(LookerRefinementResolver.REFINEMENT_PREFIX)
+
+    @staticmethod
+    def merge_column(
+        original_dict: dict, refinement_dict: dict, key: str
+    ) -> List[dict]:
+        """
+        Check refinement parameter override at https://help.looker.com/hc/en-us/articles/4419773929107-LookML-refinements
+        """
+        merge_column: List[dict] = []
+        original_value: Any = (
+            original_dict.get(key) if original_dict.get(key) is not None else []
+        )
+        refine_value: Any = (
+            refinement_dict.get(key) if refinement_dict.get(key) is not None else []
+        )
+        # name is required field, not going to be None
+        original_column_map = {
+            column[LookerRefinementResolver.NAME]: column for column in original_value
+        }
+        refine_column_map = {
+            column[LookerRefinementResolver.NAME]: column for column in refine_value
+        }
+        for existing_column_name in original_column_map:
+            existing_column = original_column_map[existing_column_name]
+            refine_column = refine_column_map.get(existing_column_name)
+            if refine_column is not None:
+                existing_column.update(refine_column)
+                # delete the entry from refine_column_map
+                del refine_column_map[existing_column_name]
+
+            merge_column.append(existing_column)
+
+        # merge any remaining column from refine_column_map
+        for new_column_name in refine_column_map:
+            merge_column.append(refine_column_map[new_column_name])
+
+        return merge_column
+
+    @staticmethod
+    def merge_and_set_column(
+        new_raw_view: dict, refinement_view: dict, key: str
+    ) -> None:
+        merged_column = LookerRefinementResolver.merge_column(
+            new_raw_view, refinement_view, key
+        )
+        if merged_column:
+            new_raw_view[key] = merged_column
+
+    @staticmethod
+    def merge_refinement(raw_view: dict, refinement_views: List[dict]) -> dict:
```

Comment:
please add a docstring comment for this method

Reply:
done